tests/test_plugin.py

```diff
@@ -407,43 +427,58 @@ def test_menu_cache_invalidate_after_menucontent_archive(self):
         # NOTE: This test is based on a similar one from django-cms:
         # https://github.com/divio/django-cms/blob/2daeb7d63cb5fee49575a834d0f23669ce46144e/cms/tests/test_plugins.py#L160
 
-        # Set up a versioned page with one placeholder
-        page_content = factories.PageContentWithVersionFactory(
-            language=self.language, version__created_by=self.get_superuser()
+        menu = factories.MenuFactory()
```

Comment:
This test is now broken, they are very different to the other tests because they are testing logic that happens when you actually publish. The factories fake the transition of Draft -> Published -> Draft by just getting to the state directly. When you actually .publish() you are running the logic that happens on publish. It's important that this test works as it did before by running the publish method.

Reply:
There should be one menucontent and one pagecontent for this test. Menu content will start published because we are testing the page publishing and the menu archiving!

```
menucontent = factories.MenuContentWithVersionFactory(language=self.language, version__state=PUBLISHED,)
pagecontent = factories.PageContentWithVersionFactory(language=self.language, version__created_by=self.get_superuser())
```